presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1378,6 +1388,57 @@ static protocol::Duration getUptime(
   return protocol::Duration(seconds, protocol::TimeUnit::SECONDS);
 }
 
+void PrestoServer::convertToVeloxPlan(
+    proxygen::HTTPMessage* message,
+    proxygen::ResponseHandler* downstream,
+    const std::vector<std::unique_ptr<folly::IOBuf>>& body) {
+  protocol::ExecutionFailureInfo failure;
+  try {
+    auto headers = message->getHeaders();
+
+    std::ostringstream oss;
+    for (auto& buf : body) {
+      oss << std::string((const char*)buf->data(), buf->length());
+    }
+    std::string planFragmentJson = oss.str();
+    protocol::PlanFragment planFragment = json::parse(planFragmentJson);
+
+    auto queryCtx = core::QueryCtx::create();
+    VeloxInteractiveQueryPlanConverter converter(queryCtx.get(), pool_.get());
+
+    // Create static taskId and empty TableWriteInfo needed for plan conversion
+    protocol::TaskId taskId = "velox-plan-conversion.0.0.0";
+    auto tableWriteInfo = std::make_shared<protocol::TableWriteInfo>();
+
+    // Attempt to convert the plan fragment to a Velox plan
+    if (auto writeNode =
+            std::dynamic_pointer_cast<const protocol::TableWriterNode>(
+                planFragment.root)) {
+      // TableWriteInfo is not yet built at the planning stage, so we can not
+      // fully convert a TableWriteNode and skip that node of the fragment.
+      converter.toVeloxQueryPlan(writeNode->source, tableWriteInfo, taskId);
```

Comment:
@BryanCutler : You have to call the VeloxPlanValidator as well https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/types/VeloxPlanValidator.cpp#L35

We can eventually remove it after this plan checker is in production.